Toric/Mathlib/CategoryTheory/WithTerminal/Cones.lean

```diff
@@ -0,0 +1,180 @@
+    | of a, of b => (inferInstance : Fintype (a ⟶ b))
+  }
+
+  /-- For any functor `K : J ⥤ Over X`, there is a canonical extension
+  `WithTerminal J ⥤ C`, since `Over X` has a final object, `𝟙 X`. It is
+  easier to define it ad hoc, since the final object can be chosen canonically
+  (note that any isomorphism mapping to `X` is also final)-/
+  @[simps!]
+  def liftFromOver {X : C} (K : J ⥤ Over X) : WithTerminal J ⥤ C :=
+    WithTerminal.lift
+      (K ⋙ Over.forget X)
+      (fun a ↦ (K.obj a).hom)
+      (by aesop_cat)
+
+  /-- The extension of a functor to over categories behaves well with compositions -/
+  @[simps]
+  def extendCompose {X : C} (K : J ⥤ Over X) (F : C ⥤ D) :
+   (liftFromOver K ⋙ F) ≅ liftFromOver (K ⋙ (Over.post F)) where
+    hom := {
+      app
+      | star => 𝟙 _
+      | of a => 𝟙 _
+    }
+    inv := {
+      app
+      | star => 𝟙 _
+      | of a => 𝟙 _
+    }
+
+  @[simps]
+  def coneLift {X : C} {K : J ⥤ Over X} : Cone K ⥤ Cone (liftFromOver K) where
+  obj t := {
+    pt := t.pt.left
+    π := {
+      app a := match a with
+      | of a => CommaMorphism.left (t.π.app a)
+      | star => t.pt.hom
+      naturality a b f:=
+      match a, b with
+      | star , star
+      | of a, star => by aesop
+      | star, of _ => by contradiction
+      | of a, of b => by
+        have : (t.π.app b).left = (t.π.app a).left ≫ (K.map f).left := by
+          calc
+            (t.π.app b).left = (t.π.app a ≫ K.map f).left := by
+              simp only [Functor.const_obj_obj, Cone.w]
+            _ = (t.π.app a).left ≫ (K.map f).left := rfl
+        simpa [Functor.const_obj_obj, Cone.w]
+    }
+  }
+  map {t₁ t₂} f := {
+    hom := f.hom.left
+    w := by
+      intro a
+      cases a with
+      | star => aesop_cat
+      | of a =>
+          have := by calc
+            f.hom.left ≫ (t₂.π.app a).left =  (f.hom ≫ t₂.π.app a).left := by rfl_cat
+            _ = (t₁.π.app a).left := by simp_all only [ConeMorphism.w, Functor.const_obj_obj]
+          simpa
+  }
+
+  @[simps]
+  def coneBack {X : C} {K : J ⥤ Over X} : Cone (liftFromOver K) ⥤ Cone K where
+  obj t := {
+    pt := Over.mk (t.π.app star)
+    π := {
+        app a := {
+            left := t.π.app (of a)
+            right := 𝟙 _
+            w := by
+              have := by
+                calc
+                  t.π.app (of a) ≫ (K.obj a).hom = t.π.app (of a)  ≫
+                    (liftFromOver K).map (homFrom a) := rfl
+                  _ = t.π.app star := by simp only [Functor.const_obj_obj, Cone.w]
+              simp [this]
+        }
+        naturality a b f := by
+          ext
+          let f₂ := incl.map f
+          have eq_after_K: (K.map f₂).left = (K.map f).left := by aesop
+          have nat : t.π.app (of b) =
+            t.π.app (of a) ≫ (K.map f₂).left := by simpa using t.π.naturality f₂
+          simp [nat, eq_after_K]
+    }
+  }
+  map {t₁ t₂ f} := {
+    hom := Over.homMk f.hom (by aesop_cat)
```

Comment:
A lot of constructions in category are designed to discharge side goals using `aesop_cat`, so no need to call it explicitly
```suggestion
    hom := Over.homMk f.hom
```